### PR TITLE
feat/allowing multiple tags for operation meta

### LINKED ISSUE #82: fix/support multiple tags per endpoint
I was trying out the package and noticed that each endpoint is limited to a **single tag**.
Please see below.

```javascript
const testRouter = createRouter().query("hello", {
	meta: { openapi: { enabled: true, method: "GET", path: "/hello", tag: "OnlyASingleTag" } },
	input: z.object({ /** input stuff */ }),
	output: z.object({ /** output stuff */ }),
	async resolve({ input }) {
		/** implementation stuff */
	},
});
```
Maybe this needs to change? Since tags are used to group operations together and a single one could be in two groups. Also, Swagger Docs on the OpenAPI Spec for 3.0.3, states that the **tag** field should container an **Array of string tags. -> string[]**
[https://swagger.io/specification/#operation-object](https://swagger.io/specification/#operation-object)

### PR DESCRIPTION
**Pull request by SeanCassiere**
Closes #82 
Old PR: #83

### CODE REVIEW

File: src/generator/paths.ts [hunk 1]
```diff
@@ -81,7 +81,7 @@ export const getOpenApiPathsObject = (
           operationId: mutationPath,
           summary,
           description,
-          tags: tag ? [tag] : undefined,
+          tags: tags ? tags : tag ? [tag] : undefined,
```
Comment (jlalmes) [inline src/generator/paths.ts:84]: Nit: nested ternaries are confusing 😅

```suggestion
          tags: tags ?? tag ? [tag] : undefined,
```

File: src/generator/paths.ts [hunk 2]
```diff
@@ -40,7 +40,7 @@ export const getOpenApiPathsObject = (
           operationId: queryPath,
           summary,
           description,
-          tags: tag ? [tag] : undefined,
+          tags: tags ? tags : tag ? [tag] : undefined,
```
Comment (jlalmes) [inline src/generator/paths.ts:43]: Same as below.

**Review by jlalmes — CHANGES_REQUESTED** [review]

**Comment by SeanCassiere** [comment]
Sure thing, made the change and currently, a test is failing. Will try and fix it and commit it.

Will tag after.

File: src/generator/paths.ts (hunk 1 above)
Comment (SeanCassiere) [reply src/generator/paths.ts:84]: Will push a fresh commit

**Comment by SeanCassiere** [comment]
@jlalmes Houston we have a problem. Things get weird with the nullish coalescing operator.
image
To escape the ternary tree, will have to introduce if statements as a utility or something to that effect to resolve the Tags.

**Comment by jlalmes** [comment]
This seems to work 👉 `tags ?? (tag ? [tag] : undefined)`

**Review by jlalmes — APPROVED** [review]
LGTM - Ship it! ✅

**Comment by dodas** [comment]
@jlalmes we originally had `tags: string[]` instead of `tag: string`, right?
Just wondering what was the reason to remove it and now reverting it.

**Comment by jlalmes** [comment]
@dodas It was me seeking a "better" DX, because it's very rare that I ever add more than one tag to an endpoint. I suppose having both is a good solution for the time-being.

**Comment by SeanCassiere** [comment]
@jlalmes Squash and Merge? or is there a release pipeline/strategy for this?

**Comment by jlalmes** [comment]
@SeanCassiere squash please! I will manually release after you've merged 👍

**Comment by SeanCassiere** [comment]
> Release under `trpc-openapi@0.4.0`, thanks for your hard work @SeanCassiere 🚀

Thanks a bunch for all your help. Was my second contribution to OSS and y'all made it great.

**Pull request merged**